src/bmp.c

```diff
@@ -4,14 +4,16 @@
 
 #include "bmp.h"
 
+char new_filename[100];
+
 // Internal function of this .c file (translation unit)
 static int round4(int x) {
     return x % 4 == 0 ? x : x - x % 4 + 4;
 }
 
 void write_bmp(const char *filename, const char *rgb, int length, int width) {
     int height = (length / 3) / width;
-
+    int screenshots_count = 0;

```

Comment:
You forgot to make this static. This will make every screenshot override the previous.